processing/src/main/java/org/apache/druid/query/rowsandcols/ArrayListRowsAndColumns.java

```diff
@@ -212,15 +212,8 @@ public <T> T as(Class<T> clazz)
   @Override
   public void addColumn(String name, Column column)
   {
-    if (rows.size() == numRows()) {
-      extraColumns.put(name, column);
-      columnNames.add(name);
-      return;
-    }
-
```

Comment:
instead of removing it entirely - possibly a check would be nice to ensure that only columns which `ColumnValueSwapper` gets added like  that.

Reply:
made it check for instances of ObjectArrayColumn